node/src/Pos/Explorer/Txp/Toil/Logic.hs

```diff
@@ -165,14 +165,11 @@ delTxExtraWithHistory id addrs = do
 
 updateUtxoSumFromBalanceUpdate :: MonadTxExtra m => BalanceUpdate -> m ()
 updateUtxoSumFromBalanceUpdate balanceUpdate = do
-    let plusChange  = foldr unsafeAddCoin (mkCoin 0) $ map snd $ plusBalance  balanceUpdate
-        minusChange = foldr unsafeAddCoin (mkCoin 0) $ map snd $ minusBalance balanceUpdate
-        utxoChange  =
-            if plusChange > minusChange then
-                (Plus, unsafeSubCoin plusChange minusChange)
-            else
-                (Minus, unsafeSubCoin minusChange plusChange)
-    uncurry updateUtxoSum utxoChange
+    let plusChange  = sum $ map (coinToInteger . snd) $ plusBalance  balanceUpdate
+        minusChange = sum $ map (coinToInteger . snd) $ minusBalance balanceUpdate
+        utxoChange  = plusChange - minusChange
+    utxoSum <- getUtxoSum
+    putUtxoSum $ utxoSum + utxoChange
```

Comment:
Ok, that's somewhat simpler. I'd probably write it like this:
```
let utxoChange = sum $ [   coinToInteger coin  | (_, coin) <- plusBalance  balanceUpdate ]
                    ++ [ -(coinToInteger coin) | (_, coin) <- minusBalance balanceUpdate ]
```

Also, why do we keep the +ve and -ve balances separately in `BalanceUpdate`?